.github/workflows/tests.yml

```diff
@@ -110,90 +140,127 @@ jobs:
 
     - name: Install Ingress
       run: |
-        kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v${{ matrix.ingress }}/deploy/static/provider/baremetal/deploy.yaml
-        sleep 5
-        time kubectl -n ingress-nginx wait --for=condition=Available deployment --all --timeout 300s
-
-    - uses: actions/checkout@v2
-    - name: go tests
-      run: |
-        make go-test
-
-    - name: fetch logs
-      if: ${{ failure() }}
-      run: |
-        mkdir  -p /tmp/logs
-        kind export logs --name harbor /tmp/logs
-
-    - uses: actions/upload-artifact@v2
-      if: ${{ failure() }}
-      with:
-        name: kind
-        path: /tmp/logs
+        kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v${{ matrix.ingress }}/deploy/static/provider/kind/deploy.yaml
+        time kubectl wait --namespace ingress-nginx --for=condition=ready pod --selector=app.kubernetes.io/component=controller --timeout=300s
 
     - name: build harbor-operator
       run: |
-        make docker-build IMG=${dockerImage}
+        make manifests docker-build IMG=${dockerImage}
         kind load docker-image ${dockerImage} --name harbor
 
     - name: install harbor-operator
       run: |
-        set -ex
-        make helm-install NAMESPACE="${operatorNamespace}" IMG=${dockerImage}
-        kubectl -n "${operatorNamespace}" wait --for=condition=Available deployment --all --timeout 300s
+        cd manifests/cluster
+        kustomize edit add secret github-token --disableNameSuffixHash --from-literal=GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
+        kustomize edit add patch --path patch/github-token.yaml
+        kustomize edit set image goharbor/harbor-operator=${dockerImage}
+        kustomize build | kubectl apply -f -
+
+        # make helm-install IMG=${dockerImage}
+
+        if ! time kubectl -n ${operatorNamespace} wait --for=condition=Available deployment --all --timeout 300s; then
+          kubectl get all -n ${operatorNamespace}
+          exit 1
+        fi
 
     - name: install harbor
       run: |
-        export GITHUB_TOKEN=xxx
         set -ex
-        make sample
-        for i in $(seq 1 6);do
+        IP=`hostname -I | awk '{print $1}'`
+        echo "IP=$IP" >> $GITHUB_ENV
+        CORE_HOST=core.$IP.nip.io
+        NOTARY_HOST=notary.$IP.nip.io
+        echo "CORE_HOST=$CORE_HOST" >> $GITHUB_ENV
+        echo "NOTARY_HOST=$NOTARY_HOST" >> $GITHUB_ENV
+        sed -i "s/core.harbor.domain/$CORE_HOST/g" manifests/samples/full_stack_other_fs.yaml
+        sed -i "s/notary.harbor.domain/$NOTARY_HOST/g" manifests/samples/full_stack_other_fs.yaml
+        sed -i "s/logLevel: info/logLevel: debug/g" manifests/samples/full_stack_other_fs.yaml
```

Comment:
Looks like a bit hacky.
The target `make sample` does not work anymore?
I suggest to add few parameters: `make sample CORE_HOST=$CORE_HOST NOTARY_HOST=$NOTARY_HOST`, wdyt?

Reply:
`make sample` is still using old Harbor CR, not using HarborCluster CR yet, correctly we are focus on testing with yamls in folder manifests